### PR TITLE
Fix secondary master to create checkpoints

### LINKED ISSUE #8390: secondary_master does not create checkpoints
**Alluxio Version:** 1.8.0, 1.8.1

**Describe the bug**
The secondary_master process doesn't create checkpoints like it's supposed to.

**To Reproduce**
configure
```
alluxio.master.journal.checkpoint.period.entries=1
alluxio.master.journal.log.size.bytes.max=10
```
in `alluxio-site.properties`, then run a master and secondary_master
```
$ bin/alluxio-start.sh master
$ bin/alluxio-start.sh secondary_master
$ bin/alluxio-start.sh worker
```

After running `bin/alluxio runTests`, the secondary_master should have created some checkpoints in `journal/FileSystemMaster/v1/checkpoints`.

**Expected behavior**
Expected the secondary master to create checkpoints

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by aaudiber**
Previously, it wouldn't do anything. Now it creates
checkpoints like it's supposed to.

fixes #8390

### CODE REVIEW

File: tests/src/test/java/alluxio/server/ft/journal/MultiMasterJournalTest.java [hunk 1]
```diff
@@ -48,17 +41,7 @@ public void testCheckpointReplay() throws Exception {
...
-    CommonUtils.waitFor("checkpoint to be written", () -> {
-      UfsJournalSnapshot snapshot;
-      try {
-        snapshot = UfsJournalSnapshot.getSnapshot(journal);
-      } catch (IOException e) {
-        throw new RuntimeException(e);
-      }
-      return !snapshot.getCheckpoints().isEmpty();
-    });
+    IntegrationTestUtils.waitForCheckpoint(Constants.FILE_SYSTEM_MASTER_NAME);
```
Comment (gpang) [inline tests/src/test/java/alluxio/server/ft/journal/MultiMasterJournalTest.java]: How come this didn't fail before?

File: core/server/master/src/main/java/alluxio/master/AlluxioSecondaryMaster.java [hunk 2]
```diff
@@ -77,19 +82,30 @@
 
   @Override
   public void start() throws Exception {
-    mRegistry.start(false);
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/AlluxioSecondaryMaster.java]: we don't need to update the `mRegistry`?

File: tests/src/test/java/alluxio/server/ft/journal/MultiMasterJournalTest.java (hunk 1 above)
Comment (aaudiber) [reply tests/src/test/java/alluxio/server/ft/journal/MultiMasterJournalTest.java]: this is a multi-master test, so there isn't any secondary master involved. Backup masters work corectly

File: core/server/master/src/main/java/alluxio/master/AlluxioSecondaryMaster.java (hunk 2 above)
Comment (aaudiber) [reply core/server/master/src/main/java/alluxio/master/AlluxioSecondaryMaster.java]: secondary_master never serves, so it doesn't need to start the masters. The *only* thing that secondary_master ever does is replay the journal, write checkpoints, and delete old journal files

**Pull request merged**